Camille Marchet
============

----

I am a CNRS researcher in sequence bioinformatics. My topics are data structures for indexing sequencing data at scale, and algorithms to work with sequences, in particular _de novo_ and RNA-seq long and short reads.

----

# Education

* 2015-2018 PhD in Computer Science, University of Rennes (France). Direction: Pierre Peterlongo (IRISA/INRIA): From reads to transcripts: _de novo_ methods for the analysis of transcriptome second and third generation sequencing.

* 2013 MS in Ecology Evolution Biometry**, Claude Bernad University of Lyon (France)
* 2013 Engineer degree in Bioinformatics and modeling, INSA Lyon (France)
* 2007 Baccalauréat, Lycée Romain Rolland (Clamecy, France)


# Professional career

* 2021-present, Chargée de recherche CNRS (junior researcher) in BONSAI team, CRIStAL lab, Lille, France
* 2022-2023 and 2020-2021 Career breaks (parental leave)
* 2018 visiting in Zamin Iqbal's group, Cambridge
* 2018-2020 Postdoc in BONSAI team, CRIStAL lab, Lille, France (supervisors: Rayan Chikhi, Mikaël Salson)
* 2015-2018 PhD student in GenScale team, IRISA lab, Rennes, France (director: Pierre Peterlongo)
* 2013-2015 Research engineer in ERABLE team, LBBE lab, Lyon, France (supervisor: Vincent Lacroix)

# Teaching, training and dissemination of science
 
* **PhD students (as a co-supervisor)**
	- 2021-present Khodor Hannoush, subject: Dynamic pangenome graphs, with Pierre Peterlongo (IRISA Rennes, France)
	- 2022-present Thomas Baudeau, subject: Mapping methods for new long reads viral data, with Mikaël Salson (CRIStAL Lille, France)
	- 2023-present Igor Martayan, subject: Data structures for efficient operations on sequence sets, with Jean-Stéphane Varré (CRIStAL Lille, France)


* **Teaching**
	- 2019, 2021, 2023: Data structures (MISO Masters, Bioinformatics, University of Lille)
	- 2019, 2020: Algorithmics and data structures (Licence 2, Computer Science, University of Lille)
	- 2017: Modeling and engineering for the living (Masters, Computer Science, INSA Rennes)
	- 2016: Data bases (Licence 2, INSA Rennes)
	- 2015, 2016: Biostatistics, R programming (Licence 3, Mathematics/Comp Sci, ENS Rennes)
	- 2014: Mathematics for biology (Licence 1, Biology, Lyon University Claude Bernard)


* **Training**
	- 2020, 2022, 2023, 2024: Teacher at Evomics Workshop on Genomics (Czech Republic)
	- 2021: Teacher at (JC)2BIM, GDR BIM's school on algorithmics and statistics for bioinformatics (France)
	- 2019, 2021-23: courses on RNA-seq analysis for Bilille and CNRS (France)


* **Popularization**
	- 2020: Two press articles for Binaire of Le Monde journal
	- 2015: Genome Assembly workshop, IRISA Lab open day (France)

* **Organization of scientific events**
	- SPIRE (2021), JOBIM (2022), SeqBIM (2023)


# Research administration and management

* **PhD defense jurys**
	- 2023: Victor Epain, _Assemblage de fragments ADN : structures de graphes et échafaudage de génomes de chloroplastes_ (directed by Rumen Andonov), IRISA Rennes (France)
	- 2021: Claudio Lorenzi, _Design and implementation of bioinformatic tools for RNA sequencing data analysis_ (directed by William RITCHIE & Alban MANCHERON), IGH Montpellier (France)



# Editorial/program committees


* **Reviewing for journals, conferences**
	- Regular reviews for Nature Methods, Bioinformatics, Nature Communication, NAR, RECOMB, ISMB/ECCB, RECOMB-seq

* **Program committee for conferences/workshops**
	- ISBM (2022), WABI (2022), ECCB (2022), SPIRE (2020), RECOMB-seq (2020, 2018)




# Research


* **Selected Publications since 2015**
	- [A survey of mapping algorithms in the long-reads era](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-023-02972-3), K Sahlin, T Baudeau, B Cazaux & C Marchet; _Genome Biology_ **2023**
	- [Scalable sequence database search using Partitioned Aggregated Bloom Comb-Trees](https://www.biorxiv.org/content/10.1101/2022.02.11.480089v2), C Marchet & A Limasset; _ISMB_ **2023**
	- [Fractional Hitting Sets for Efficient and Lightweight Genomic Data Sketching](https://www.biorxiv.org/content/10.1101/2023.06.21.545875v1), T Rouzé, I Martayan, C Marchet & A Limasset, _WABI_ 2023
	- [BLight: Efficient exact associative structure for k-mers](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab217/6209734), **C Marchet**, M Kerbiriou, A Limasset; _Bioinformatics_, **2021**
	- REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets, **C Marchet**, Z Iqbal, D Gautheret, M Salson, R Chikhi; _Bioinformatics_, **2020**
	- Data structures based on k-mers for querying large collections of sequencing data sets, **C Marchet**, C Boucher, SJ Puglisi, P Medvedev, M Salson, R Chikhi; _Genome Research_, **2020**

* **Software**
	- [https://transipedia.fr](https://transipedia.fr) and [github.com/kamimrcht/](github.com/kamimrcht/)

* **Involvement in research projects since 2018**

	- 2024-.. Find-RNA ANR,_Data-structures for collections of sets of (meta-)transcriptomics data_(coordinator)
	- 2023-.. Full-RNA ANR, _Indexing large scale 2nd and 3rd generation RNA datasets_(member)
	- 2022-.. INSSANE ANR, _Novel methods for studying RNA structures_(responsible for Univ Lille)
	- 2021-.. ALPACA ITN, _Methods for pangenomics_(member)
	- 2019-.. Seqdigger ANR, _Indexing large scale genomic samples"_ (member)
	- 2018-2022 Transipédia ANR, _Indexing large scale RNA-seq datasets__(member)

* **Communication**

- Selection of invited talks since 2017
	- 2023: Hashing-based data-structures for querying large k-mer (collections of) sets, CiE, Batumi (Georgia)
	- 2023: Hashing-based data-structures for querying large k-mer (collections of) sets, Sequences in London (United Kingddom)
	- 2022: Scalable sequence database search using approximate membership data structures, Genome Informatics, Cambridge (United Kingdom)
	- 2022: How to improve student/advisor relationships, WABI, ALPACA 2nd Annual Workshop, Potsdam (Germany)
	- 2020: From reads to transcripts: _de novo_ methods for the analysis of transcriptome second and third generation sequencing, SIF congress, INSA de Lyon (France)

- Selection of talks in workshops/conferences since 2015
	- 2020: REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets, ISMB, Montreal (Canada, virtual conference)
	- 2019: Indexing De Bruijn graphs with minimizers, RECOMB-seq, Washington DC (US)
	- 2019: Indexing De Bruijn graphs with minimizers, BiATA, St Petersburg (Russia)
	- 2019: Survey of _k_-mer set of sets data structures for querying large collections of sequencing datasets, Helsinki Bioinformatics Day (Finland)
	- 2018: CARNAC-LR : Clustering coefficient-based Acquisition of RNA Communities in Long Reads, JOBIM, Marseille (France)

# Research contracts

* 2024-2028, as a coordinator: ANR project Find-RNA, ~190k euros, collaboration with CRIL, University of Lens and SciLife lab, University of Stockholm
* 2022-2026, as a partner: ANR project INSSANE, ~330k euros, collaboration with LIX Polytechnique, CiTCoM Université de Paris and LCBPT Université de Paris

# Prizes, awards, fellowships

* 2019: Honorable mention for the Gilles Kahn prize of the Société Informatique de France for PhDs in Computer Science
