Camille Marchet
============

----

I am a CNRS researcher in sequence bioinformatics. My topics are data structures for indexing sequencing data at scale, and algorithms to work with sequences, in particular _de novo_ and RNA-seq long and short reads.

----

Education
---------

2015-2018
: **PhD in Computer Science**, University of Rennes (France). Thesis title: _From reads to transcripts: de novo methods for the analysis of transcriptome second and third generation sequencing._ Jury:
:	Eric Coissac (MCF, LECA) - Rapporteur
:	Hélène Touzet (DR, CRISTAL) - Rapportrice
:	Thomas Derrien (CR, IGDR) - Rapporteur
:	Dominique Lavenier (DR, IRISA) - Examinateur
:	Thierry Lecroq, (PR, LITIS) - Examinateur
:	Pierre Peterlongo, (CR, IRISA) - Directeur
:	Anne Siegel, (DR, IRISA) - Examinatrice
:	Hagen Tilgner, (Lecturer, Weil Cornell NY) - Examinateur

2013
: **MS in Ecology Evolution Biometry**, Claude Bernad University of Lyon (France)
: **Engineer degree in Bioinformatics and modeling**, INSA Lyon (France)

2007
: **Baccalauréat**, Lycée Romain Rolland (Clamecy, France)


Professional background
----------
**Chargée de recherche CNRS in BONSAI team, CRIStAL lab, Lille, France (2021-..)**

**Career break (sept 2020-sept 2021)**
Maternity leave followed by parental leave.

**Postdoc in BONSAI team, CRIStAL lab, Lille, France (2018-2020)** (supervisors: Rayan Chikhi, Mikaël Salson)

**PhD student in GenScale team, IRISA lab, Rennes, France (2015-2018)** (director: Pierre Peterlongo)
[Link to my thesis](https://tel.archives-ouvertes.fr/tel-01939193/)

**Research engineer in ERABLE team, LBBE lab, Lyon, France (2013-2015)** (supervisor: Vincent Lacroix)

Collaborations
--------------
- 2019 Indexing techniques for pangenomes and datasets, Iqbal lab, EBI Cambridge, UK
- 2018-.. Scalable  of indexing RNA-seq dataset abundances, INSERM Montpellier, France and Orsay University, France
- 2018 Polishing RNA long reads, Tilgner lab, Wellcome Trust NY, US
- 2018-19 Multiple sequence alignment for long read correction, TIBS team, University of Normandy, France
- 2017-19 Correction of RNA Nanopore long reads, BONSAI team, Lille, France and ERABLE team, Lyon, France
- 2017-18 Clustering of RNA Nanopore long reads, Genoscope, Evry, France
- 2016-18 Indexing techniques for de novo meta-transcriptomics, Sorbonne University Paris, France
- 2013-15 _De novo_ identification of SNP in RNA-seq for eco-evolution, LBBE Lyon, France
- 2013-15 _De novo_ identification of alternative splicing events in RNA-seq, ENS Lyon, France


Publications
------------

**Journal articles**

* [BLight: Efficient exact associative structure for k-mers](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab217/6209734), **C Marchet**, M Kerbiriou, A Limasset; _Bioinformatics_, **2021**
* [Scalable long read self-correction and assembly polishing with multiple sequence alignment](https://www.nature.com/articles/s41598-020-80757-5)
P Morisse, **C Marchet**, A Limasset, T Lecroq, A Lefebvre; _Scientific reports 11_, **2021**
* [REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets](https://academic.oup.com/bioinformatics/article/36/Supplement_1/i177/5870500?login=true), **C Marchet**, Z Iqbal, D Gautheret, M Salson, R Chikhi; _Bioinformatics_, **2020**
* [Data structures based on k-mers for querying large collections of sequencing data sets](https://scholar.google.com/citations?view_op=view_citation&hl=fr&user=q07hs7YAAAAJ&citation_for_view=q07hs7YAAAAJ:Xz60mAmATU4C), **C Marchet**, C Boucher, SJ Puglisi, P Medvedev, M Salson, R Chikhi; _Genome Research_, **2020**
* [ELECTOR: Evaluator for long reads correction methods](https://www.biorxiv.org/content/10.1101/512889v1), **C Marchet**, P Morisse, L Lecompte, A Limasset, A Lefebvre, T Lecroq, P Peterlongo; _Nucleic Acids Research Genomics and Bioinformatics_, **2019**
* [Clustering _de Novo_ by Gene of Long Reads from Transcriptomics Data](https://academic.oup.com/nar/article/47/1/e2/5107577), **C Marchet**, L Lecompte, C Da Silva, C Cruaud, J-M Aury, J Nicolas, P Peterlongo; _Nucleic Acids Research_, **2018**
* [A _de novo_ approach to disentangle partner identity and function in holobiont systems](https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0481-9), A Meng**+**, **C Marchet+**, E Corre, P Peterlongo, A Alberti, C Da Silva, P Wincker, E Pelletier, I Probert, J Decelle, S Le Crom, F Not, L Bittner; _Microbiome_, **2018** (**+ co-first authors**)
* [A resource-frugal probabilistic dictionary and applications in bioinformatics](https://www.sciencedirect.com/science/article/pii/S0166218X18301288), C Marchet, L Lecompte, A Limasset, L Bittner, P Peterlongo; _Discrete Applied Mathematics_, **2018**
* [Comparative assessment of long-read error-correction software applied to RNA-sequencing data](https://academic.oup.com/bib/advance-article-abstract/doi/10.1093/bib/bbz058/5512144), L Ishi Soares de Lima, **C Marchet**, S Caboche, C Da Silva, B Istace, J-M Aury, H Touzet, R Chikhi; _Briefings in Bioinformatics_, **2019**
* [SNP calling from RNA-seq data without a reference genome: identification, quantification, differential analysis and impact on the protein sequence](https://www.ncbi.nlm.nih.gov/pubmed/27458203), H Lopez Maestre, L Brinza, **C Marchet**, J Kielbassa, S Bastien, M Boutigny, D Monnin, A El Filali, CM Carareto, C Vieira, F Picard, N Kremer, F Vavre, M-F Sagot, V Lacroix; _Nucleic Acids Research_, **2016**
* [Complementarity of assembly-first and mapping-first approaches for alternative splicing annotation and differential analysis from RNAseq data](https://www.nature.com/articles/s41598-018-21770-7), C Benoit-Pilven, **C Marchet**, E Chautard, L Lima, M-P Lambert, G Sacomoto, A Rey, C Bourgeois, D Auboeuf, V Lacroix; _Scientific Reports_, **2018**
* [Playing hide and seek with repeats in local and global de novo transcriptome assembly of short RNA-seq reads](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5322684/), L Lima, B Sinaimeri, G Sacomoto, H Lopez-Maestre, C Marchet, V Miele, M-F Sagot and V Lacroix; _Algorithms for Molecular Biology_, **2017**
* [Colibread on galaxy: a tools suite dedicated to biological information extraction from raw NGS reads](https://gigascience.biomedcentral.com/articles/10.1186/s13742-015-0105-2), E Rivals, A Andrieux, A Z El Aabidine, B Cazaux, **C Marchet**, C Lemaitre, C Monjeaud, G Sacomoto, L Salmela, O Collin, P Peterlongo, R Uricaru, S Alves-Carvalho, V Lacroix, V Miele, Y LeBras; _GigaScience_, **2016**

**Papers accepted to conferences with proceedings**

* [REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets](https://academic.oup.com/bioinformatics/article/36/Supplement_1/i177/5870500?login=true), C Marchet, Z Iqbal, D Gautheret, M Salson, R Chikhi; _ISMB_, **2020**
* [Indexing De Bruijn graphs with minimizers](https://www.biorxiv.org/content/10.1101/546309v2), **C Marchet**, A Kerbiriou, A Limasset, _RECOMB-seq_, **2019**
* [CONSENT: Scalable self-correction of long reads with multiple sequence alignment
](https://www.biorxiv.org/content/10.1101/546630v3), P Morisse, **C Marchet**, A Limasset, A Lefebvre, T Lecroq, _RECOMB-Seq_, **2019**
* [A resource-frugal probabilistic dictionary and applications in (meta)genomics](https://hal.inria.fr/hal-01386744/document), **C Marchet**, A Limasset, L Bittner, P Peterlongo, _PSC_, **2016**
* [Navigating in a Sea of Repeats in RNA-seq without Drowning](https://link.springer.com/chapter/10.1007/978-3-662-44753-6_7), G Sacomoto, B Sinaimeri, **C Marchet**, V Miele, MF Sagot, V Lacroix, International Workshop on Algorithms in Bioinformatics (WABI), **2014**


Software
---------

* [My Github](github.com/kamimrcht)
* [Kissplice](kissplice.prabi.fr)
* [KissDE](bioconductor.org/packages/release/bioc/html/kissDE.html)

Communications
--------------

**Invited talks to international workshops/conferences**

- 2018 A _de novo_ approach to disentangle partner identity and function in holobiont systems, Advances techniques to study and exploit the sponge and coral microbiomes Workshop, ULB Brussels (Belgium)

**Invited talks to national workshops/conferences**

- 2020 From reads to transcripts: _de novo_ methods for the analysis of transcriptome second and third generation sequencing, SIF congress, INSA de Lyon (France)
- 2020 Scalable data structures for sequencing data, Symposium GDR Madics, Lyon (France) (covid-19 : canceled)
- 2018 CARNAC-LR and C2C: de novo clustering and detection of alternative isoforms in Third Generation Sequencing transcriptomes, [Genotoul Biostats/Bioinfo day](https://bioinfo-biostat.sciencesconf.org/), INRA Toulouse (France)
- 2017 _De novo_ Clustering of Gene Expressed Variants in Transcriptomic Long Reads Data Sets at workshop RNA-Seq and Nanopore sequencing, Genoscope Evry (France)

**Invited talks to lab seminars**

- 2019 New methodologies for the analysis of transcriptome sequences, MAB team seminar, LIRMM Montpellier (France)
- 2018 From reads to transcripts: _de novo_ methods to analyze transcriptome 2d and 3d generations sequencing data, Roscoff Biological Station seminar (France)
- 2018 A highly scalable data structure for read similarity computation and its application to marine holobionts, EEB group meeting, ULB Brussels (Belgium)
- 2018 CARNAC-LR: clustering genes expressed variants from long read RNA sequencing, team TIBS seminar, LITIS Rouen (France)
- 2017 Rconnector: a resource-frugal probabilistic dictionary and applications in (meta)genomics and transcriptomics, LBBE NGS group seminar, Lyon (France)

**Selected talks in workshops/conferences**

- 2020 [REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets](http://biata2020.spbu.ru/), BIATA, St Petersburg (Russia, virtual conference)
- 2020 [REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets](https://www.iscb.org/cms_addon/conferences/ismb2020/proceedings.php#HitSeq:%20High-throughput%20Sequencing), ISMB, Montreal (Canada, virtual conference)
- 2020 [REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets](https://jobim2020.sciencesconf.org/program), JOBIM, Montpellier (France, virtual conference)
- 2020 REINDEER: efficient indexing of k-mer presence and abundance in sequencing datasets, DSB, Rennes (France)
- 2019 [Indexing De Bruijn graphs with minimizers](https://recombseq.recomb2019.org/schedule/#Sat), RECOMB-seq, Washington DC (US)
- 2019 [Indexing De Bruijn graphs with minimizers](http://biata2019.spbu.ru/program/), BiATA, St Petersburg (Russia)
- 2019 [Survey of _k_-mer set of sets data structures for querying large collections of sequencing datasets](https://dsb2019.gitlab.io/), DSB, Dortmund (Germany)
- 2019 [Survey of _k_-mer set of sets data structures for querying large collections of sequencing datasets]( https://www.helsinki.fi/en/researchgroups/bioinformatics/bioinformatics-day-2019), Helsinki Bioinformatics Day (Finland)
- 2019 Read correction for non-uniform coverages, [RCAM](http://maiage.jouy.inra.fr/?q=fr/rcam2019) Institut Pasteur Paris (France)
- 2019 Survey of _k_-mer set of sets data structures for querying large collections of sequencing datasets, seqBIM Marne la Vallée (France)
- 2018 [BCOOL-Trans: accurate and variant-preserving correction for RNA-seq](http://bioinfo.univ-rouen.fr/seqbio2018/programme.php), Seqbio, Rouen (France)
- 2018 [ELECTOR: EvaLuator of Error CorrectionTools for lOng Reads](http://bioinfo.univ-rouen.fr/seqbio2018/programme.php), Seqbio, Rouen (France)
- 2018 [CARNAC-LR : Clustering coefficient-based Acquisition of RNA Communities in Long Reads](https://jobim2018.sciencesconf.org/data/pages/Program_A4_JOBIM2018.pdf), JOBIM, Marseille (France)
- 2017 _De novo_ Clustering of Gene Expression in Transcriptomic Long Reads Data Sets, Seqbio, Lille (France)
- 2017 A highly scalable data structure for read similarity computation and its application to marine holobionts, RCAM, Paris (France)
- 2017 A scaling transcriptomic approach to study holobiont data sets, 4e colloque Génomique Environnementale, Marseille (France)
- 2016 Rconnector: a resource-frugal probabilistic dictionary and applications in (meta)genomics and transcriptomics, SeqBio, Nantes (France)
- 2015 kissDE : a replicate-wise and annotation-free R package for testing the association between differential variants and experimental conditions in high throughput sequencing data, SeqBio 2015, Orsay (France)
- 2015 kissDE : a package to test for differences in reads counts derived from variants in RNA sequencing data, Quatriemes rencontres R 2015, Lyon (France)

Scientific life
----------------

**Abroad visits**

- 2019 (3 months) Iqbal lab, EBI, (Cambridge, UK)
- 2015 (3 weeks) Laboratório Nacional de Computação Científica (LNC, Petrópolis, Brazil) and Universidade de São Paulo (USP, Brazil)

**Involvement in scientific events**

- 2020 PC, SPIRE
- 2020 PC, RECOMB-seq (Italy)
- 2020 PC, seqBIM (France)
- 2018 Volunteer at RECOMB-seq and RECOMB (France)
- 2016 Organisation commitee for “Biological insights from raw high-throughput sequencing data” - Colib’read ANR workshop (France)

**Participation in research projects**

- 2020 Seqdigger ANR project (large scale indexation of sets of datasets)
- 2018-20 Transipédia ANR project (index data structures for collections of RNA-seq data)
- 2018-20 ASTER ANR project (algorithms for RNA Nanopore reads)
- 2017 MASTODONS CNRS project (algorithms for long read correction)
- 2015-18 Hydrogen ANR project (indexing methods for environmental meta-genomics)
- 2013-16 Colib'read ANR project (De novo methods for variant calling in NGS data)

**Training for scientists/staff**

- 2020 Teacher at Evomics Workshop on Genomics (Czech Republic)
- 2019, 2021 Teacher at Bilille training courses on RNA-seq analysis. The course I made with Pierre Pericard is [here](https://wikis.univ-lille.fr/bilille/_media/ngs2019-rna-seq.pdf) (France)
- 2015 Tutor at CNRS course: “Bioinformatique pour les NGS” (France)
- 2014 Tutor at BGE & EMBnet tutorials: “RNA-seq analysis” (France)
- 2014 Tutor at PRABI training courses, “Analyse de données RNA-seq sous l’environnement Galaxy” (France)

**Vulgarization**
- 2021 [Press article](https://www.lemonde.fr/blog/binaire/2020/05/06/sars-cov-2-et-covid-19-on-va-jouer-sur-les-mots/) ,a short popularization article about coronavirus assembly with long reads (in French)
- 2020 [Press article](https://www.lemonde.fr/blog/binaire/2020/02/07/rencontre-a-la-frontiere-entre-linformatique-et-la-biologie/) about my PhD thesis (in French)
- 2015 Genome Assembly workshop, IRISA Lab open day (France)
- 2014-15 Introduction to academic jobs to undergrad students (INSA Lyon, France)


**Involvement in lab and team's lifes**

- 2019 BONSAI team's lab day organization (Lille, France)
- 2018-19 Male/female parity committee IRISA lab (Rennes, France)
- 2017 GenScale team's retreat organization (Rennes, France)
- 2017 Animation of a working group on long read correction (Rennes, France)



Teaching
--------
2019/2020
: Lille University, informatics (license 2): algorithmics and data structures
2018/2019
: Lille University, informatics (license 2): algorithmics and data structures

2016/2017
: INSA Rennes engineer school, informatics (master): modeling and engineering for the living
: INSA Rennes engineer school, informatics (license 2): data bases
: ENS Rennes, informatics/mathematics (license 3): biostatistics, R programming

2015/2016
: ENS Rennes, informatics/mathematics (license 3): biostatistics, R programming

2014/2015
: Lyon University Claude Bernard, biology (license 1): mathematics for biology

Supervision
-------------
* 2021-.. Khodor Hannoush (PhD student), subject: Dynamic pangenome graphs. Co-supervised with Pierre Peterlongo (Inria Rennes)
* 2020 Agathénaïs Adiguna (L3 internship), subject: Data-structures for large scale RNA queries. Co-supervised with Rayan Chikhi (Institut Pasteur)
* 2018 Benjamin Churcheward (Master 2 internship), subject: Isoform prediction using long reads sequencing. Co-supervised with Pierre Peterlongo (Inria Rennes)
* 2017 Lolita Lecompte (Master 2 internship), subject: Conception and evaluation of a pipeline for _de novo_ study of long reads in transcriptomics. Co-supervised with Pierre Peterlongo (Inria Rennes)
* 2014 Camille Sessegolo (License 3 internship), subject: Improvements in the kissplice2refgenome software. Co-supervised with Vincent Lacroix (Université Claude Bernard Lyon 1)




Reviews for journals/conferences
--------------------------------
* 2020 ISMB/ECCB, BMC Supplements, Bioinformatics, Nucleic Acids Research, Gigascience, RECOMB-Seq, RECOMB, Bioinformatics, SPIRE
* 2019 BMC Supplements, Bioinformatics, RECOMB, Nucleic Acid Research
* 2018 GigaScience
* 2017 BMC Bioinformatics

Funding
------
* 2021-.. ANR project INSSANE, as a partner, collaboration with LIX Polytechnique, CiTCoM Université de Paris and LCBPT Université de Paris: integration  of  novel  algorithms,  custom  chemistry,  and  new  probing  protocols  for  studying large structured RNA

Awards, fellowships
---------------
* 2019 Honorable mention for the Gilles Kahn prize of the Société Informatique de France for PhDs in Computer Science
* 2020 SFBI travel grant
